### PR TITLE
Add compiled version & git hash to version command

### LINKED ISSUE #84: Add git commit to version output
Would be nice to know which version is actually being used apart from the hardcoded `v0.0.1`.

Maybe something like 
```
boulder, version 0.0.1 (git abf3456)
```

### PR DESCRIPTION
**Pull request by tarkah**
Resolves #84 

I'm not super familiar with `meson`, so I'm not sure if this is the best way to pass these values in. 

This adds a new `environment` module which contains two new environment variables that are populated during compile by meson. 

`GIT_VERSION` won't be present when compiling from a tarball / archive, only from the repo. Formatted version takes this into account and only includes it if available. The git hash also includes `-dirty` if the repo is dirty.

See output below of compiling on clean branch, dirty branch and after removing `.git` folder. 

image

### CODE REVIEW

**Comment by livingsilver94** [comment]
Serpent OS supports both Meson and Dub as build systems. It is my understanding here that you only addressed Meson.

**Comment by tarkah** [comment]
I've landed on an implementation that works for both `dub` and `meson`.

I'm not 100% thrilled with the implementation but I couldn't figure out a better way to handle this within the constraints of build build systems.

Note there seems to be an issue with `meson` where if `./build/environment` file exists, it'll use that as the string import instead of the file placed under `build`. I couldn't figure out how to prevent this with `meson`. 

image

File: dub.json [hunk 1]
```diff
@@ -18,5 +18,13 @@
 	},
 	"stringImportPaths": [
 		"data"
-	]
+	],
+	"preBuildCommands": [
+		"cp data/environment.in data/environment",
+		"sed -i 's/@VERSION@/'$DUB_PACKAGE_VERSION'/g' data/environment",
+		"sed -i 's/@GIT_HASH@/'`git describe --tags --always --dirty --abbrev=7 2>/dev/null || echo ''`'/g' data/environment",
+	],
```
Comment (tarkah) [inline dub.json:26]: `dub` appears to prevent us from assigning to intermediary variables :shrug:

**Comment by ermo** [comment]
@tarkah Any chance for a rebase?

@Ikey I think we should get this in as soon as the rebase is done?

**Comment by tarkah** [comment]
@ermo I've rebased and made a change to exclude tags, since it's overly verbose as we separately show version:

### Before

`boulder, version 1.0.1 (v1.0.1-6-gf879fb5)`

### After

`boulder, version 1.0.1 (52c8064)`
`boulder, version 1.0.1 (f879fb5-dirty)`

**Pull request merged**